Create glossary.md (#7)
Open-Attestation/adr: glossary.md

# Glossary of Terms

OpenAttestation: The underlying technology/framework that we use to power verifiable credentials

OpenCerts: The project that is owned by SSG and is concerned with educational credentials

TradeTrust: The project that is owned by IMDA and is concerned with trade documents as well as transferable asset deeds

### OpenAttestation Framework
  

>   Envelope: Wrapper around a document that uses OpenAttestation framework to protect its integrity. _(Sometimes called Signed JSON file)_

Reviewer: WrappedDocument ?

  
  Proof: Part of the `Envelope` metadata, a set of Keccak256 hashes that leads to the Merkle root that is notarised (i.e on the Ethereum blockchain)
  
  Document: The information inside the .opencert or .tt file that excludes the `Envelope` metadata. _(Sometimes called Unsigned JSON)_
  
  Schema: Refers to a specific schema that the contained `Document` has to adhere to.
  

>   Template: Instructions on how to display the data in the `Document`, could be a URL pointing to a `Renderer` or simply a path depending on OpenAttestation version

Reviewer 2: ```suggestion
  Template: Instructions on how to display the data in the `Document`, it is currently an URL pointing to a `Renderer`. For OpenCerts, a simple path is used for v1.4 & v1.5, prior to that, a key is used.  
```

Reviewer 2: We have two legacy in OpenCerts. 
Also wanted to show that going with the path/key is **not the norm**

>   Renderer: A HTTP(s) endpoint that takes a `Document` and returns some browser-renderable contents (i.e HTML/CSS/JS)

Reviewer 2: ```suggestion
  Renderer: A website to be embedded in the OpenAttestation platform. It should take a `Document` and render it in browser. A sample application is available in the `decentralized-renderer` repository.
```

Reviewer 2: Wanted to highlight that is is actually rendered.

Author: technically the browser renders it, the url just takes the document and returns some code for the browser to render

>   RendererFrame: A component on the OpenCerts/OpenAttestation site that takes a `Document` and sends it to the `Renderer` endpoint.

Reviewer 2: This is confusing, people don't need to know about this also.

### OpenCerts

>   SSG: SkillsFuture Singapore, the owner of OpenCerts

Reviewer 2: ```suggestion
  SSG: SkillsFuture Singapore, the owner of OpenCerts accreditation registry
```

>   SSG Registry: A list of entities considered to be trustworth issuers, administered by SSG _(Sometimes called The Registry or just Registry)_

Reviewer 2: ```suggestion
  SSG Registry: A list of accredited institutes, administered by SSG _(Sometimes called The Registry or just Registry)_
```

  
  Issuer: An entity that issues educational credentials, represented by one or more DocumentStores.
  
  Certificate: .OpenCert file that contains a verifiable educational credential.
  
  DocumentStore: An Ethereum smart contract that keeps track of OpenCerts issuances and revocations.
  

>   DocumentStore Address: The specific Ethereum Address that a `DocumentStore` is located at.

Reviewer 2: Wanna differentiate with the wallet address ?
